test/run-functional-tests

```diff
@@ -0,0 +1,419 @@
+#!/bin/bash
+
+DOCKER_IMAGE=${DOCKER_IMAGE:-"liangxin1300/hatbw"}
+DOCKER_SERVICE="docker.service"
+COROSYNC_CONF="/etc/corosync/corosync.conf"
+COROSYNC_AUTH="/etc/corosync/authkey"
+HA_NETWORK_FIRST="ha_network_first"
+HA_NETWORK_SECOND="ha_network_second"
+declare -a HA_NETWORK_ARRAY
+declare -a HA_NETWORK_V6_ARRAY
+HA_NETWORK_ARRAY[0]=$HA_NETWORK_FIRST
+HA_NETWORK_ARRAY[1]=$HA_NETWORK_SECOND
+HA_NETWORK_V6_ARRAY[0]="2001:db8:10::/64"
+HA_NETWORK_V6_ARRAY[1]="2001:db8:20::/64"
+BEHAVE_CASE_DIR="$(dirname $0)/features/"
+BEHAVE_CASE_EXCLUDE="sbd|ocfs2"
+read -r -d '' COROSYNC_CONF_TEMPLATE << EOM
+totem {
+        version: 2
+        cluster_name: hacluster
+        clear_node_high_bit: yes
+        interface {
+                ringnumber: 0
+                mcastport: 5405
+                ttl: 1
+        }
+
+        transport: udpu
+        crypto_hash: sha1
+        crypto_cipher: aes256
+        token: 5000
+        join: 60
+        max_messages: 20
+        token_retransmits_before_loss_const: 10
+}
+
+logging {
+        fileline: off
+        to_stderr: no
+        to_logfile: no
+        logfile: /var/log/cluster/corosync.log
+        to_syslog: yes
+        debug: off
+        timestamp: on
+        logger_subsys {
+                subsys: QUORUM
+                debug: off
+        }
+
+}
+
+nodelist {
+}
+
+quorum {
+
+        # Enable and configure quorum subsystem (default: off)
+        # see also corosync.conf.5 and votequorum.5
+        provider: corosync_votequorum
+}
+EOM
+
+
+fatal() {
+	error $*
+	exit 1
+}
+
+
+error() {
+	echo "ERROR: $*"
+}
+
+
+warning() {
+	echo "WARNING: $*"
+}
+
+
+info() {
+	echo "INFO: $*"
+}
+
+
+is_number() {
+	num=$1
+	test ! -z "$num" && test "$num" -eq "$num" 2> /dev/null && test "$num" -gt 0 2> /dev/null
+}
+
+
+check_docker_env() {
+	# check if docker available
+	systemctl list-unit-files $DOCKER_SERVICE &> /dev/null
+	if [ "$?" -ne 0 ];then
+		fatal "$DOCKER_SERVICE is not available"
+	fi
+	# check if docker.service started
+	systemctl is-active $DOCKER_SERVICE &> /dev/null
+	if [ "$?" -ne 0 ];then
+		fatal "$DOCKER_SERVICE is not active"
+	fi
+	# check if docker cgroup driver is systemd
+	docker info 2> /dev/null|grep -q "Cgroup Driver: systemd"
+	if [ "$?" -ne 0 ];then
+		warning "docker cgroup driver suggest to be \"systemd\""
+	fi
```

Comment:
And the further check is expected to detect if the host is the cgroup2 environment. That says, this only works for cgroup2 so far. And, currently, it works for Tumbelweed and 15.4+ host. This should return the fatal error if try on 15.3, for example.

Reply:
I've already tried this on 15SP3, basically, it works well, although not sure which test case might failed on it, that's the reason we make this warning:
```
WARNING: docker cgroup driver suggest to be "systemd"
```

Reply:
I read the misleading here. My intention was not for the code to detect "Cgroup Driver". That's fine.

I should put the comment in the beginning of check_docker_env(). The point is to clarify with the user currently this tool does not support cgroup v1 host, since it need --hostns stuff as for a different `docker_options` later on.

Reply:
Anyway, most of functionalities seems work now even on Cgroup Driver
It make sense here just warning not error and exit

Reply:
I see the misreading still.

Here is the sample code in my head
```
[ $(findmnt /sys/fs/cgroup -no FSTYPE) == "cgroup2" ] ||  { echo Exit. cgroup2 is not detected; exit; }
```

Reply:
It looks I have some missing docker knowledge between cgroup v1 and v2 in the different distro. I agree to leave it as it is for the time being, and revisit when it gets crystal clear.